Test crashes when InputDecoration.enabledBorder property is set to null

## Steps to Reproduce

Test below crashes with this error:
```
Testing started at 15:39 ...
/Users/ts/Documents/flutter/bin/flutter --no-color test --machine test/crashing_example_test.dart
Shell: ../../third_party/skia/src/core/SkGeometry.cpp:1458: fatal error: "assert(0 <= dot && dot <= 1.0f + (1.0f / (1 << 12)))"dart:async/stream_controller.dart 595:43                     _StreamController.addError
dart:async/stream_controller.dart 862:13                     _StreamSinkWrapper.addError
package:stream_channel/src/guarantee_channel.dart 140:14     _GuaranteeSink._addError
package:stream_channel/src/guarantee_channel.dart 131:5      _GuaranteeSink.addError
package:flutter_tools/src/test/flutter_platform.dart 576:29  _FlutterPlatform._startTest.<fn>
===== asynchronous gap ===========================
dart:async/zone.dart 1053:19                                 _CustomZone.registerUnaryCallback
dart:async/runtime/libasync_patch.dart 77:23                 _asyncThenWrapperHelper
package:flutter_tools/src/test/flutter_platform.dart         _FlutterPlatform._startTest.<fn>
package:flutter_tools/src/test/flutter_platform.dart 782:26  _FlutterPlatform._startTest
===== asynchronous gap ===========================
dart:async/zone.dart 1053:19                                 _CustomZone.registerUnaryCallback
dart:async/runtime/libasync_patch.dart 77:23                 _asyncThenWrapperHelper
package:flutter_tools/src/test/flutter_platform.dart         _FlutterPlatform._startTest
package:flutter_tools/src/test/flutter_platform.dart 462:36  _FlutterPlatform.loadChannel
package:flutter_tools/src/test/flutter_platform.dart 417:46  _FlutterPlatform.load
===== asynchronous gap ===========================
dart:async/zone.dart 1053:19                                 _CustomZone.registerUnaryCallback
dart:async/runtime/libasync_patch.dart 77:23                 _asyncThenWrapperHelper
package:test_core/src/runner/loader.dart                     Loader.loadFile.<fn>
package:test_core/src/runner/load_suite.dart 98:31           new LoadSuite.<fn>.<fn>
===== asynchronous gap ===========================
dart:async/zone.dart 1045:19                                 _CustomZone.registerCallback
dart:async/zone.dart 962:22                                  _CustomZone.bindCallbackGuarded
dart:async/timer.dart 52:45                                  new Timer
dart:async/timer.dart 87:9                                   Timer.run
dart:async/future.dart 174:11                                new Future
package:test_api/src/backend/invoker.dart 399:21             Invoker._onRun.<fn>.<fn>.<fn>

Shell subprocess crashed with unexpected exit code -5 after tests finished.
Test: /Users/ts/Documents/[path to project]/crashing_example_test.dart
Shell: /Users/ts/Documents/flutter/bin/cache/artifacts/engine/darwin-x64/flutter_tester
```

```dart
import 'dart:async';

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  group('Clickable input field', () {
    testWidgets(
        'on pressed isFocused should be true, after pressed finishes should become false again',
        (tester) async {
      final completer = Completer();
      Future<void> pressedFunction() => completer.future;

      await tester.pumpWidget(
        MaterialApp(
          home: Material(
            child: ClickableInputField(
              onPressed: pressedFunction,
            ),
          ),
        ),
      );

      await tester.pump();

      await tester.tap(find.byType(InkWell));

      await tester.pumpAndSettle();

      final ClickableInputFieldState state =
          tester.state(find.byType(ClickableInputField));

      expect(state.isFocused, isTrue);

      completer.complete();

      await tester.pumpAndSettle();

      expect(state.isFocused, isFalse);
    });
  });
}

typedef InputFieldPressedCallback = Future<void> Function();

class ClickableInputField extends StatefulWidget {
  const ClickableInputField({
    Key key,
    @required this.onPressed,
  }) : super(key: key);

  final InputFieldPressedCallback onPressed;

  @override
  ClickableInputFieldState createState() => ClickableInputFieldState();
}

class ClickableInputFieldState extends State<ClickableInputField> {
  bool isFocused = false;

  Future<void> onInputPress() async {
    setState(() {
      isFocused = true;
    });

    await widget.onPressed();

    removeFocus();
  }

  void removeFocus() => setState(() {
        isFocused = false;
      });

  @override
  Widget build(BuildContext context) => InkWell(
        onTap: onInputPress,
        child: Theme(
          data: Theme.of(context).copyWith(
            hintColor: isFocused ? Theme.of(context).accentColor : null,
          ),
          child: InputDecorator(
            decoration: InputDecoration(
              labelText: 'Test',
              labelStyle: isFocused
                  ? TextStyle(color: Theme.of(context).accentColor)
                  : null,
              enabledBorder: isFocused
                  ? OutlineInputBorder(
                      borderSide:
                          BorderSide(color: Theme.of(context).accentColor),
                    )
                  : null,
            ),
            child: const Text('Value'),
          ),
        ),
      );
}
```
## Logs

```
▶ flutter doctor -v
[✓] Flutter (Channel beta, v1.2.1, on Mac OS X 10.14.3 18D109, locale en-LT)
    • Flutter version 1.2.1 at /Users/ts/Documents/flutter
    • Framework revision 8661d8aecd (2 weeks ago), 2019-02-14 19:19:53 -0800
    • Engine revision 3757390fa4
    • Dart version 2.1.2 (build 2.1.2-dev.0.0 0a7dcf17eb)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/ts/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[!] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    ✗ Verify that all connected devices have been paired with this computer in Xcode.
      If all devices have been paired, libimobiledevice and ideviceinstaller may require updating.
      To update with Brew, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew uninstall --ignore-dependencies usbmuxd
        brew install --HEAD usbmuxd
        brew unlink usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 1.9.4
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 33.3.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] VS Code (version 1.31.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.23.0

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

! Doctor found issues in 1 category.
```

--- a/packages/flutter/lib/src/material/input_border.dart
+++ b/packages/flutter/lib/src/material/input_border.dart
@@ -412,7 +412,7 @@ class OutlineInputBorder extends InputBorder {
 
     const double cornerArcSweep = math.pi / 2.0;
     final double tlCornerArcSweep = start < center.tlRadiusX
-      ? math.asin(start / center.tlRadiusX)
+      ? math.asin((start / center.tlRadiusX).clamp(-1.0, 1.0))
       : math.pi / 2.0;
 
     final Path path = Path()
@@ -474,12 +474,12 @@ class OutlineInputBorder extends InputBorder {
       final double extent = lerpDouble(0.0, gapExtent + gapPadding * 2.0, gapPercentage);
       switch (textDirection) {
         case TextDirection.rtl: {
-          final Path path = _gapBorderPath(canvas, center, gapStart + gapPadding - extent, extent);
+          final Path path = _gapBorderPath(canvas, center, math.max(0.0, gapStart + gapPadding - extent), extent);
           canvas.drawPath(path, paint);
           break;
         }
         case TextDirection.ltr: {
-          final Path path = _gapBorderPath(canvas, center, gapStart - gapPadding, extent);
+          final Path path = _gapBorderPath(canvas, center, math.max(0.0, gapStart - gapPadding), extent);
           canvas.drawPath(path, paint);
           break;
         }

--- a/packages/flutter/test/material/input_decorator_test.dart
+++ b/packages/flutter/test/material/input_decorator_test.dart
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
 import 'dart:io' show Platform;
 
 import 'package:flutter/material.dart';
@@ -2253,6 +2254,41 @@ void main() {
     expect(getBorderRadius(tester), BorderRadius.zero);
     await tester.pumpAndSettle();
     expect(getBorderRadius(tester), BorderRadius.zero);
+  });
+
+  testWidgets('OutlineInputBorder async lerp', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/28724
+
+    final Completer<void> completer = Completer<void>();
+    bool waitIsOver = false;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return GestureDetector(
+              onTap: () async {
+                setState(() { waitIsOver = true; });
+                await completer.future;
+                setState(() { waitIsOver = false;  });
+              },
+              child: InputDecorator(
+                decoration: InputDecoration(
+                  labelText: 'Test',
+                  enabledBorder: !waitIsOver ? null : const OutlineInputBorder(borderSide: BorderSide(color: Colors.blue)),
+                ),
+              ),
+            );
+          },
+        ),
+      ),
+    );
+
+    await tester.tap(find.byType(StatefulBuilder));
+    await tester.pumpAndSettle();
+
+    completer.complete();
+    await tester.pumpAndSettle();
   });
 
   test('InputBorder equality', () {

